### PR TITLE
[seal] Update to SEAL 3.6.1

### PR DESCRIPTION
**Pull request by kiromaru**
Update SEAL port to version 3.6.1

- What does your PR fix? Updating to a new SEAL version

- Which triplets are supported/not supported? Have you updated the CI baseline?
Tested x64-windows-static, x64-linux, x64-osx
x64-windows/x86-windows are not supported as the library only supports static linkage in Windows.

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
Yes

### CODE REVIEW

File: ports/seal/CONTROL [hunk 1]
```diff
@@ -1,7 +1,8 @@
 Source: seal
-Version: 3.6.0
+Version: 3.6.1
 Homepage: https://github.com/microsoft/SEAL
 Description: Microsoft SEAL is an easy-to-use and powerful homomorphic encryption library.
+Supports: !windows | (windows & static)
```
Comment (JackBoosY) [inline ports/seal/CONTROL:5]: I think the master version support non-windows, why drop *inx support here?

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/seal/CONTROL (hunk 1 above)
Comment (kiromaru) [reply ports/seal/CONTROL:5]: We are not dropping it... the expression specifies that the library supports all operating systems other than Windows (`!windows`) or specifically Windows with static linkage (`windows & static`).

Comment (JackBoosY) [reply ports/seal/CONTROL:5]: Oh, I'm blind...

Comment (JackBoosY) [reply ports/seal/CONTROL:5]: If seal only support dynamic build on Windows, please add `vcpkg_check_linkage(ONLY_DYNAMIC_LIBRARY)` to the top of `portfile.cmake`.

Comment (kiromaru) [reply ports/seal/CONTROL:5]: SEAL supports only static linkage on Windows. The `portfile.cmake` file already contains the line `vcpkg_check_linkage(ONLY_STATIC_LIBRARY)`.

**Review by JackBoosY — APPROVED** [review]

**Comment by JonLiu1993** [comment]
@kiromaru ,The feature test on windows is successful, but there seems to be a problem with x64-linux on linux, can you take a look?
````
home/vzhli17/Jon/test2/vcpkg/buildtrees/seal/src/d394eb2aeb-d50aefddbc.clean/native/src/seal/util/gcc.h:10:19: error: SEAL requires __GNUC__ >= 6
 #pragma GCC error "SEAL requires __GNUC__ >= 6"
                   ^
In file included from /home/vzhli17/Jon/test2/vcpkg/buildtrees/seal/src/d394eb2aeb-d50aefddbc.clean/native/src/seal/memorymanager.h:6:0,
                 from /home/vzhli17/Jon/test2/vcpkg/buildtrees/seal/src/d394eb2aeb-d50aefddbc.clean/native/src/seal/encryptionparams.h:6,
                 from /home/vzhli17/Jon/test2/vcpkg/buildtrees/seal/src/d394eb2aeb-d50aefddbc.clean/native/src/seal/context.h:6,
                 from /home/vzhli17/Jon/test2/vcpkg/buildtrees/seal/src/d394eb2aeb-d50aefddbc.clean/native/src/seal/ciphertext.h:6,
                 from /home/vzhli17/Jon/test2/vcpkg/buildtrees/seal/src/d394eb2aeb-d50aefddbc.clean/native/src/seal/ciphertext.cpp:4:
/home/vzhli17/Jon/test2/vcpkg/buildtrees/seal/src/d394eb2aeb-d50aefddbc.clean/native/src/seal/util/defines.h:120:28: error: ‘byte’ in namespace ‘std’ does not name a type
     using seal_byte = std::byte;
                            ^
In file included from /home/vzhli17/Jon/test2/vcpkg/buildtrees/seal/src/d394eb2aeb-d50aefddbc.clean/native/src/seal/util/globals.h:6:0,
                 from /home/vzhli17/Jon/test2/vcpkg/buildtrees/seal/src/d394eb2aeb-d50aefddbc.clean/native/src/seal/memorymanager.h:7,
````
````
/home/vzhli17/Jon/test2/vcpkg/buildtrees/seal/src/d394eb2aeb-d50aefddbc.clean/native/src/seal/util/common.h: In function ‘constexpr T seal::util::mul_safe(T, T)’:
/home/vzhli17/Jon/test2/vcpkg/buildtrees/seal/src/d394eb2aeb-d50aefddbc.clean/native/src/seal/util/defines.h:133:30: error: expected ‘(’ before ‘constexpr’
 #define SEAL_IF_CONSTEXPR if constexpr
                              ^
/home/vzhli17/Jon/test2/vcpkg/buildtrees/seal/src/d394eb2aeb-d50aefddbc.clean/native/src/seal/util/common.h:140:13: note: in expansion of macro ‘SEAL_IF_CONSTEXPR’
             SEAL_IF_CONSTEXPR(std::is_unsigned<T>::value)
             ^
/home/vzhli17/Jon/test2/vcpkg/buildtrees/seal/src/d394eb2aeb-d50aefddbc.clean/native/src/seal/ciphertext.cpp:347:1: error: expected ‘}’ at end of input
 } // namespace seal
 ^
/home/vzhli17/Jon/test2/vcpkg/buildtrees/seal/src/d394eb2aeb-d50aefddbc.clean/native/src/seal/ciphertext.cpp: At global scope:
/home/vzhli17/Jon/test2/vcpkg/buildtrees/seal/src/d394eb2aeb-d50aefddbc.clean/native/src/seal/ciphertext.cpp:347:1: error: expected ‘}’ at end of input
/home/vzhli17/Jon/test2/vcpkg/buildtrees/seal/src/d394eb2aeb-d50aefddbc.clean/native/src/seal/ciphertext.cpp:347:1: error: expected ‘}’ at end of input
[2/40] /usr/bin/c++  -I/home/vzhli17/Jon/test2/vcpkg/buildtrees/seal/src/d394eb2aeb-d50aefddbc.clean/native/src -Inative/src -isystem /home/vzhli17/Jon/test2/vcpkg/installed/x64-linux/include -fPIC -Wall -Wextra -Wconversion -Wshadow -pedantic -g -fPIC -pthread -std=gnu++1z -MD -MT CMakeFiles/seal.dir/native/src/seal/context.cpp.o -MF CMakeFiles/seal.dir/native/src/seal/context.cpp.o.d -o CMakeFiles/seal.dir/native/src/seal/context.cpp.o -c /home/vzhli17/Jon/test2/vcpkg/buildtrees/seal/src/d394eb2aeb-d50aefddbc.clean/native/src/seal/context.cpp
FAILED: CMakeFiles/seal.dir/native/src/seal/context.cpp.o
/usr/bin/c++  -I/home/vzhli17/Jon/test2/vcpkg/buildtrees/seal/src/d394eb2aeb-d50aefddbc.clean/native/src -Inative/src -isystem /home/vzhli17/Jon/test2/vcpkg/installed/x64-linux/include -fPIC -Wall -Wextra -Wconversion -Wshadow -pedantic -g -fPIC -pthread -std=gnu++1z -MD -MT CMakeFiles/seal.dir/native/src/seal/context.cpp.o -MF CMakeFiles/seal.dir/native/src/seal/context.cpp.o.d -o CMakeFiles/seal.dir/native/src/seal/context.cpp.o -c /home/vzhli17/Jon/test2/vcpkg/buildtrees/seal/src/d394eb2aeb-d50aefddbc.clean/native/src/seal/context.cpp
In file included from /home/vzhli17/Jon/test2/vcpkg/buildtrees/seal/src/d394eb2aeb-d50aefddbc.clean/native/src/seal/util/defines.h:106:0,
                 from /home/vzhli17/Jon/test2/vcpkg/buildtrees/seal/src/d394eb2aeb-d50aefddbc.clean/native/src/seal/memorymanager.h:6,
                 from /home/vzhli17/Jon/test2/vcpkg/buildtrees/seal/src/d394eb2aeb-d50aefddbc.clean/native/src/seal/encryptionparams.h:6,
                 from /home/vzhli17/Jon/test2/vcpkg/buildtrees/seal/src/d394eb2aeb-d50aefddbc.clean/native/src/seal/context.h:6,
                 from /home/vzhli17/Jon/test2/vcpkg/buildtrees/seal/src/d394eb2aeb-d50aefddbc.clean/native/src/seal/context.cpp:4:
/home/vzhli17/Jon/test2/vcpkg/buildtrees/seal/src/d394eb2aeb-d50aefddbc.clean/native/src/seal/util/gcc.h:10:19: error: SEAL requires __GNUC__ >= 6
 #pragma GCC error "SEAL requires __GNUC__ >= 6"
                   ^
In file included from /home/vzhli17/Jon/test2/vcpkg/buildtrees/seal/src/d394eb2aeb-d50aefddbc.clean/native/src/seal/memorymanager.h:6:0,
                 from /home/vzhli17/Jon/test2/vcpkg/buildtrees/seal/src/d394eb2aeb-d50aefddbc.clean/native/src/seal/encryptionparams.h:6,
                 from /home/vzhli17/Jon/test2/vcpkg/buildtrees/seal/src/d394eb2aeb-d50aefddbc.clean/native/src/seal/context.h:6,

````

**Comment by kiromaru** [comment]
Hi JonLiu1993,
SEAL requires at least GCC 7, that is the cause of the error. Notice that the PR pipeline was able to build x64-linux without problems. I tested it locally as well.

**Pull request merged**